lib/text_parser.dart

```diff
@@ -12,7 +12,7 @@ List<TextSegment> parseText(String? text) {
       unicode: true);
   final matches = exp.allMatches(text);
   var start = 0;
-  matches.forEach((match) {
+  for (var match in matches) {
```

Comment:
Please add some detail for words matches logic

Reply:
`// it means that there is some text before matched word e.g text = some text some text [matched word]
    if (match.start > start) {
      // add this text to [segments] e.g add "some text some text"
      if (segments.isNotEmpty && segments.last.isText) {
        segments.last.text += text.substring(start, match.start);
      } else {
        segments.add(TextSegment(text.substring(start, match.start)));
      }
      // set the [start] to the point where match was found 
      start = match.start;
    }`